go/arrow/table.go

```diff
@@ -69,7 +69,7 @@ type Column struct {
 // of the ref counting.
 func NewColumnFromArr(field Field, arr Array) Column {
 	if !TypeEqual(field.Type, arr.DataType()) {
-		panic("arrow/array: inconsistent data type")
+		panic(fmt.Sprintf("arrow/array: inconsistent data type %s vs %s", field.Type.String(), arr.DataType().String()))
```

Comment:
I think having more data in these types of errors is helpful because they don't seem to be day to day errors but errors occurring during development, and having the types in the message removes the 'obvious second step' (ie. running in debugger to investigate) and transforms it into a more actionable, clearer error. There's one more below in L101, also one in the parquet test suite. Easy to revert.

Reply:
I can agree that adding more information in the errors is good. It might make sense to use:

`fmt.Errorf("%w: arrow/array: inconsistent data type %s vs %s", arrow.ErrInvalid, field.Type, arr.DataType)` instead of the `fmt.Sprintf`. You also don't need to call `.String()` manually, since you're using `%s` it'll use the `.String()` method automatically when printing it. So the code can look cleaner.

Reply:
same goes for the other spots.